docs/rustdoc-index.md

```diff
@@ -0,0 +1,5 @@
+# Fedimint technical reference documentation
+
+<!-- this page is used a landing page for https://docs.fedimint.org/ -->
+
+This is documentation automatically generated from the Fedimint source code.
```

Comment:
Somewhat of a placeholder for now.

Reply:
We should probably outline the structure a bit and link to the most important crates.